x-pack/plugins/observability_solution/apm/server/routes/historical_data/route.ts

```diff
@@ -30,4 +31,23 @@ const hasDataRoute = createApmServerRoute({
   },
 });
 
-export const historicalDataRouteRepository = hasDataRoute;
+const hasEntitiesRoute = createApmServerRoute({
+  endpoint: 'GET /internal/apm/has_entities',
+  options: { tags: ['access:apm'] },
+  handler: async (resources): Promise<{ hasData: boolean }> => {
+    const { context, request, logger } = resources;
+    const coreContext = await context.core;
+
+    const [entitiesESClient] = await Promise.all([
+      createEntitiesESClient({
+        request,
+        esClient: coreContext.elasticsearch.client.asCurrentUser,
+      }),
+    ]);
```

Comment:
Do we need to wrap `createEntitiesESClient` in a `Promise.all`?

Reply:
Good catch. No we don't

Reply:
fixed https://github.com/elastic/kibana/pull/188209/commits/e8e93c72efd520a40a4541a9248dbc1669257ac3